excel/src/main/java/org/apache/metamodel/excel/ExcelDataContext.java

```diff
@@ -209,13 +205,13 @@ protected void onSchemaCacheRefreshed() {
         return null;
     }
 
-    private SpreadsheetReaderDelegate getSpreadsheetReaderDelegate(Ref<InputStream> inputStream)
+    private SpreadsheetReaderDelegate getSpreadsheetReaderDelegate()
             throws MetaModelException {
         if (_spreadsheetReaderDelegate == null) {
             synchronized (this) {
                 if (_spreadsheetReaderDelegate == null) {
                     try {
-                        if (POIXMLDocument.hasOOXMLHeader(inputStream.get())) {
+                        if (POIXMLDocument.hasOOXMLHeader(getInputStream())) {
```

Comment:
Rewrite looks good, except I'm not sure I like the logger.error() when still falling back. Either cancel with an error, or try to recover with a warning. 

(this may be the same as before, but Github's diff interface is terrible on the mobile, so I can't see it)